src/twisted/conch/test/test_conch.py

```diff
@@ -547,51 +536,34 @@ def execute(self, remoteCommand, process, sshArgs=""):
 
         @return: L{defer.Deferred}
         """
-        # PubkeyAcceptedKeyTypes does not exist prior to OpenSSH 7.0 so we
-        # first need to check if we can set it. If we can, -V will just print
-        # the version without doing anything else; if we can't, we will get a
-        # configuration error.
-        d = getProcessValue(
-            which("ssh")[0], ("-o", "PubkeyAcceptedKeyTypes=ssh-dss", "-V")
-        )
-
-        def hasPAKT(status):
-            if status == 0:
-                opts = "-oPubkeyAcceptedKeyTypes=ssh-dss "
-            else:
-                opts = ""
-
-            process.deferred = defer.Deferred()
-            # Pass -F /dev/null to avoid the user's configuration file from
-            # being loaded, as it may contain settings that cause our tests to
-            # fail or hang.
-            cmdline = (
-                (
-                    "ssh -2 -l testuser -p %i "
-                    "-F /dev/null "
-                    "-oUserKnownHostsFile=kh_test "
-                    "-oPasswordAuthentication=no "
-                    # Always use the RSA key, since that's the one in kh_test.
-                    "-oHostKeyAlgorithms=ssh-rsa "
-                    "-a "
-                    "-i dsa_test "
-                )
-                + opts
-                + sshArgs
-                + " 127.0.0.1 "
-                + remoteCommand
+        process.deferred = defer.Deferred()
+        # Pass -F /dev/null to avoid the user's configuration file from
+        # being loaded, as it may contain settings that cause our tests to
+        # fail or hang.
+        cmdline = (
+            (
+                "ssh -2 -l testuser -p %i "
+                "-F /dev/null "
+                "-oUserKnownHostsFile=kh_test "
+                "-oPasswordAuthentication=no "
+                # Always use the RSA key, since that's the one in kh_test.
+                "-oHostKeyAlgorithms=ssh-rsa "
+                "-a "
+                "-i rsa_test "
             )
-            port = self.conchServer.getHost().port
-            cmds = (cmdline % port).split()
-            encodedCmds = []
-            for cmd in cmds:
-                if isinstance(cmd, str):
-                    cmd = cmd.encode("utf-8")
-                encodedCmds.append(cmd)
-            reactor.spawnProcess(process, which("ssh")[0], encodedCmds)
-            return process.deferred
-
-        return d.addCallback(hasPAKT)
+            + sshArgs
+            + " 127.0.0.1 "
+            + remoteCommand
+        )
+        port = self.conchServer.getHost().port
+        cmds = (cmdline % port).split()
+        encodedCmds = []
+        for cmd in cmds:
+            if isinstance(cmd, str):
+                cmd = cmd.encode("utf-8")
```

Comment:
```suggestion
            cmd = cmd.encode("utf-8")
```

It looks like there is a missing branch coverage here

![image](https://github.com/user-attachments/assets/d91b2810-7620-42c2-9482-2e6326e79dd8)

Reply:
I did something similar, thanks.